packages/arb-tx-aggregator/txaggregator/server.go

```diff
@@ -0,0 +1,208 @@
+	"github.com/offchainlabs/arbitrum/packages/arb-util/hashing"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/arbbridge"
+	"github.com/offchainlabs/arbitrum/packages/arb-validator-core/message"
+)
+
+const maxTransactions = 50
+
+type Server struct {
+	rollupAddress common.Address
+	globalInbox   arbbridge.GlobalInbox
+
+	sync.Mutex
+	valid        bool
+	transactions []message.Transaction
+	signatures   [][65]byte
+}
+
+// NewServer returns a new instance of the Server class
+func NewServer(
+	ctx context.Context,
+	globalInbox arbbridge.GlobalInbox,
+	rollupAddress common.Address,
+) *Server {
+	server := &Server{
+		rollupAddress: rollupAddress,
+		globalInbox:   globalInbox,
+		valid:         true,
+	}
+
+	go func() {
+		ticker := time.NewTicker(common.NewTimeBlocksInt(5).Duration())
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				server.sendBatch(ctx)
+			}
+		}
+	}()
+	return server
+}
+
+func (m *Server) sendBatch(ctx context.Context) {
+	var txes []message.Transaction
+	var sigs [][65]byte
+
+	m.Lock()
+	if !m.valid {
+		return
+	}
+
+	if len(m.transactions) == 0 {
+		m.Unlock()
+		return
+	}
+
+	if len(m.transactions) > maxTransactions {
+		txes = m.transactions[:maxTransactions]
+		sigs = m.signatures[:maxTransactions]
+		m.transactions = m.transactions[maxTransactions:]
+		m.signatures = m.signatures[maxTransactions:]
+	} else {
+		txes = m.transactions
+		m.transactions = nil
+		sigs = m.signatures
+		m.signatures = nil
+	}
+	m.Unlock()
+
+	log.Println("Submitting batch with", len(txes), "transactions")
+
+	for _, tx := range txes {
+		log.Println("tx: ", tx)
+	}
+
+	err := m.globalInbox.DeliverTransactionBatch(
+		ctx,
+		m.rollupAddress,
+		txes,
+		sigs,
+	)
+	if err != nil {
+		log.Println("Transaction aggregator failed: ", err)
+		m.Lock()
+		m.valid = false
+		m.Unlock()
+	}
+}
+
+// CallMessage takes a request from a client to process in a temporary context and return the result
+func (m *Server) SendTransaction(
+	ctx context.Context,
+	args *SendTransactionArgs,
+) (*SendTransactionReply, error) {
+	toBytes, err := hexutil.Decode(args.To)
+	if err != nil {
+		return nil, errors2.Wrap(err, "error decoding to")
+	}
+	var to common.Address
+	copy(to[:], toBytes)
+
+	sequenceNum, valid := new(big.Int).SetString(args.SequenceNum, 10)
+	if !valid {
+		return nil, errors.New("Invalid sequence num")
+	}
+
+	valueInt, valid := new(big.Int).SetString(args.Value, 10)
+	if !valid {
+		return nil, errors.New("Invalid value")
+	}
+
+	data, err := hexutil.Decode(args.Data)
+	if err != nil {
+		return nil, errors2.Wrap(err, "error decoding data")
+	}
+
+	pubkey, err := hexutil.Decode(args.Pubkey)
+	if err != nil {
+		return nil, errors2.Wrap(err, "error decoding pubkey")
+	}
+
+	signature, err := hexutil.Decode(args.Signature)
+	if err != nil {
+		return nil, errors2.Wrap(err, "error decoding signature")
+	}
+
+	if len(signature) != 65 {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '65', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)